### PR TITLE
fix(cloudProject): creation and import for cloudProject in the US

### PR DESCRIPTION
**Pull request by Stoakes**
# Description

This PR fixes public cloud project creation for OVH US. It also improves cloud project documentation so that creation in the US is easier for end user.

Because of some differences in Agora, the result is different between OVH EU/CA on one side & OVH US on the other. There's been some internal discussions about changing that (I can provide more details in PM) and they haven't landed on a result and are not expected to in any near future.

This PR works around the issue by making extra call to find out serviceName if the initial serviceName determination attempt failed.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (improve existing resource(s) or datasource(s))
- [x] Documentation update

# How Has This Been Tested?

I manually tested it by successfully creating & importing a cloud project on OVH-US.

**Test Configuration**:
* Terraform version: `terraform version`: Terraform v1.4.2
* Existing HCL configuration you used: 
``` bash
export OVH_ENDPOINT=ovh-us
```
```hcl
data "ovh_order_cart" "aaa_cart" {
  ovh_subsidiary = "us"
  description    = "us public cloud project cart"
}

data "ovh_order_cart_product_plan" "cloud" {
  cart_id        = data.ovh_order_cart.aaa_cart.id
  price_capacity = "renew"
  product        = "cloud"
  plan_code      = "project" # naming is different in the us.
}

resource "ovh_cloud_project" "cloud_project" {
  ovh_subsidiary = data.ovh_order_cart.aaa_cart.ovh_subsidiary
  description    = "us cloud project"
  payment_mean   = "ovh-account"

  plan {
    duration     = data.ovh_order_cart_product_plan.cloud.selected_price.0.duration
    plan_code    = data.ovh_order_cart_product_plan.cloud.plan_code
    pricing_mode = data.ovh_order_cart_product_plan.cloud.selected_price.0.pricing_mode
  }
}
```
And also `terraform import ovh_cloud_project.cloud_project <project_id>

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or issues
- [ ] I have added acceptance tests that prove my fix is effective or that my feature works
- [ ] New and existing acceptance tests pass locally with my changes
- [ ] I ran successfully `go mod vendor` if I added or modify `go.mod` file

### CODE REVIEW

File: website/docs/r/cloud_project.html.markdown [hunk 1]
```diff
@@ -46,7 +47,7 @@ The following arguments are supported:
 * `ovh_subsidiary` - (Required) OVHcloud Subsidiary
 * `plan` - (Required) Product Plan to order
   * `duration` - (Required) duration
-  * `plan_code` - (Required) Plan code
+  * `plan_code` - (Required) Plan code. This value must be adapted depending on your `OVH_ENDPOINT` value. It's `project.2018` for `ovh-{eu,ca}` and `project` when using `ovh-us`.`
```
Comment (scraly) [inline website/docs/r/cloud_project.html.markdown:50]:
```suggestion
  * `plan_code` - (Required) Plan code. This value must be adapted depending on your `OVH_ENDPOINT` value. It's `project.2018` for `ovh-{eu,ca}` and `project` when using `ovh-us`.
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by Stoakes** [comment]
Hello, 

Is there anything missing on this PR ? Or can we merge it ?

**Comment by scraly** [comment]
We are waiting for an US account in order to test everything as the user will do.

**Comment by Stoakes** [comment]
Creation of public cloud project patched.

The deletion (and update) of a public cloud project in the US is not working yet because `/me/notification/email/history` (used for project confirmation deletion) is not available in the US.

@yomovh & myself will see how we deal with it.

**Comment by Stoakes** [comment]
We don't want to put too much US specific logic in the provider and will work into getting resources ordered through `/order` deletable in the US.

So for now, as agreed with  @yomovh, I added a notice about the known issue that a public cloud project cannot be deleted in the US through Terraform.

**Review by yomovh — APPROVED** [review]
Thanks for your PR

**Review by scraly — APPROVED** [review]

**Comment by scraly** [comment]
After all the things we saw, edited and fixed, now, the PR is good for me 💪

**Pull request merged**